modules/core/shared/src/main/scala-3/io/circe/derivation/ConfiguredEncoder.scala

```diff
@@ -54,8 +55,17 @@ trait ConfiguredEncoder[A](using conf: Configuration) extends Encoder.AsObject[A
           JsonObject.singleton(constructorName, json)
 
 object ConfiguredEncoder:
+
+  // needed to make derivation implicitly available when summoning elemEncoders:
+  private object GivenDerivation:
+    inline given deriveEncoder[A](using conf: Configuration)(using
+      inline A: Mirror.Of[A]
+    ): Exported[ConfiguredEncoder[A]] =
+      Exported(ConfiguredEncoder.derived[A])
+
   inline final def derived[A](using conf: Configuration)(using inline mirror: Mirror.Of[A]): ConfiguredEncoder[A] =
     new ConfiguredEncoder[A] with SumOrProduct:
+      import GivenDerivation.deriveEncoder
```

Comment:
Side note: theoretically, this would allow to change the implementation of `io.circe.derivation.summonEncoder` from...
```scala
private[circe] inline final def summonEncoder[A](using Configuration): Encoder[A] =
  summonFrom {
    case encodeA: Encoder[A] => encodeA
    case _: Mirror.Of[A]     => ConfiguredEncoder.derived[A]
  }
```
...to...
```scala
private[circe] inline final def summonEncoder[A](using Configuration): Encoder[A] =
  summonInline[Encoder[A]]
```
...because derivation via `Mirror.Of[A]` is now available implictly so `summonInline[Encoder[A]]` will find it. (I tried that and all tests ran successfully.)

However: that code is also accessible via (deprecated) public API (see `io.circe.Derivation`) so I guess it's better to leave it as it is.